### PR TITLE
EOS-16520: Converting sspl dev deploy shell script to python

### PR DESCRIPTION
**Pull request by satish-seagate**
Signed-off-by: Satish Darade <satish.darade@seagate.com>

## Problem Statement
<pre>
  <code>
    Story Ref (if any):
    Please add Problem statement here...
  </code>
</pre>
## Problem Description
<pre>
  <code>
    Please add Problem decription here...
  </code>
</pre>
## Solution
<pre>
  <code>
    Please add summary of the change,List any dependencies that are required for this change.
  </code>
</pre>
## Sanity testing on RPM done
<pre>
  <code>
    - [ ] Yes
    - [ ] No
  </code>
</pre>
## Unit/Manual Testing Description
<pre>
  <code>
    Please describe the tests that you ran to verify your changes.
  </code>
</pre>

### CODE REVIEW

File: low-level/files/opt/seagate/sspl/setup/setup_yum_repos.py
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/setup_yum_repos.py:79]: why the indentation not maintained? same down below all over this script

Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/setup_yum_repos.py]: Its by VS code editor plugin which auto aligns code. 
I have corrected it now.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy]: now, I changed all references to 'LR2'.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (shriya-deshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:186]: you can use **"http://cortx-storage.colo.seagate.com/releases/cortx/components/github/main/centos-7.8.2003/dev/cortx-utils/last_successful/"** . its having recent cortx-utils pkg.

and also before installing cortx-utils , we need to install **"yum install -y gcc rpm-build python36-devel openssl-devel libffi-devel"** these pkgs. This is required for conf stor cli.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (shriya-deshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:259]: Instead of using "srvnode_1"  key directly, you can use the machine-id to fetch node_name(srvnode_1,srvnode_2 etc).

eg: machine_id=cat /etc/machine-id
srvnode=Conf.get("global_config", f"cluster>{machine_id}")

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (shriya-deshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:252]: Here you can fetch enclosure_id(enclosure_1), from  cluster>{srvnode}>storage>enclosure_id

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:62]: change later again to main once merged

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:186]: what is this special case ?

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:302]: no os.system() supposed to be used, use Simple process, this is the very reason to change it to python, to reuse code.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:180]: As commented in the script,
`# TODO: EOS-17072 - last_successful location is not having recent
# cortx-py-utils package. Untill then proceeding with static one.`
This ticket is just now resolved. we will update it.

Also the mentioned support packages are only required to build cortx-py-utils rom.
As we directly installs cortx-py-utils rpm from URL, we don't need to install these support packages I guess.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:252]: Thanks, it helps.

Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:252]: This we read from sample_global_config.yaml for single node provisioning.
Dynamically feeding machine_id and mapping a specific name to that machine_id is again a hardcoded thing at this time.
So we used to keep srvnode_1.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:245]: As discussed and as commented in previous conversation.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:62]: Sure.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:186]: 1. If found cortx-py-utils installed already when installing it from web URL, returncode is 256 stating that package already installed(Error).
2. If not installed already but installing now, returncode is 0 stating that package already installed.
`
>>> cmd
'yum install -y http://cortx-storage.colo.seagate.com/releases/eos/github/main/centos-7.8.2003/cortx-utils_99/prod/cortx_iso/cortx-py-utils-1.0.0-99_23eb199.noarch.rpm'
>>>
>>> returncode = os.system(cmd)
Loaded plugins: fastestmirror, product-id, search-disabled-repos, subscription-manager
cortx-py-utils-1.0.0-99_23eb199.noarch.rpm                                                                 | 212 kB  00:00:00
Examining /var/tmp/yum-root-wLc7tI/cortx-py-utils-1.0.0-99_23eb199.noarch.rpm: cortx-py-utils-1.0.0-99_23eb199.noarch
/var/tmp/yum-root-wLc7tI/cortx-py-utils-1.0.0-99_23eb199.noarch.rpm: does not update installed package.
Error: Nothing to do
>>>
>>> returncode
256
>>> cmd = "yum install -y cortx-py-utils"
>>> returncode = os.system(cmd)
Loaded plugins: fastestmirror, product-id, search-disabled-repos, subscription-manager
Package matching cortx-py-utils-1.0.0-83_8b3c613.noarch already installed. Checking for update.
Nothing to do
>>>
>>> returncode
0
`

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:302]: os.system() exposes stdout, stderr and rc in terminal for command execution.
But not SimpleProcess. Looking forward a way to do this.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (shriya-deshmukh) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:180]: "yum install -y gcc rpm-build python36-devel openssl-devel libffi-devel "
 if we install cortx-utils without this pkgs , then some pip dependencies will not get installed, and conf stor cli will not work. these pkgs are dependacies of pip pkgs.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (cdeshmukh) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:309]: os.system() was replaced almost in whole codebase because of security concerns. We need to avoid it.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy]: @satish-seagate , As Sandeep & Shriya mentioned, sspl-ll service uses the machine id from config file, we need to update it.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:252]: Agreed to use this way.
cc: @satish-seagate

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:180]: `TODO: EOS-17072 - last_successful location is not having recent cortx-py-utils package. Untill then proceeding with static one`
This is addressed by RE. Now last_successful is having recent cortx-py-utils RPMS.
@satish-seagate , please change this build URL and remove TODO.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:62]: @satish-seagate, This has to be take care in final commit.

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (satish-seagate) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy]: Done

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (satish-seagate) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy]: Done

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (satish-seagate) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy]: Done

**Review by shriya-deshmukh — APPROVED** [review]

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (cdeshmukh) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy:63]: after commit right!?

**Review by cdeshmukh — APPROVED** [review]

File: low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy]: No. If all good with private repo, then the final commit is to change this alone before merge.

Comment (satish-seagate) [reply low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy]: Done

**Pull request merged**
